### PR TITLE
csi: update csi sidecars' image version

### PR DESCRIPTION
**Pull request by iPraveenParihar**
Below csi sidecars are updated with latest available versions

csi-resizer: v1.11.1
csi-provisioner: v5.0.1
csi-attacher: v4.6.1
csi-snapshotter: v8.0.1
csi-node-driver-registrar: v2.11.1

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/csi/spec.go [hunk 1]
```diff
@@ -141,10 +141,10 @@ var (
 	// image names
 	DefaultCSIPluginImage   = "quay.io/cephcsi/cephcsi:v3.11.0"
 	DefaultRegistrarImage   = "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.10.1"
-	DefaultProvisionerImage = "registry.k8s.io/sig-storage/csi-provisioner:v4.0.1"
-	DefaultAttacherImage    = "registry.k8s.io/sig-storage/csi-attacher:v4.5.1"
-	DefaultSnapshotterImage = "registry.k8s.io/sig-storage/csi-snapshotter:v7.0.2"
-	DefaultResizerImage     = "registry.k8s.io/sig-storage/csi-resizer:v1.10.1"
+	DefaultProvisionerImage = "registry.k8s.io/sig-storage/csi-provisioner:v5.0.1"
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/spec.go:144]: @iPraveenParihar check do we need to disable topology for cephfs in deployment yaml.

**Comment by travisn** [comment]
@iPraveenParihar A number of the integration tests are failing with these changes. Did you have a chance to investigate the CI issues?

**Comment by iPraveenParihar** [comment]
> @iPraveenParihar A number of the integration tests are failing with these changes. Did you have a chance to investigate the CI issues?

was busy with other work and didn't get a chance to look at the failures. Will look at it today.

**Comment by Madhu-1** [comment]
@iPraveenParihar is this ready for review?

**Comment by iPraveenParihar** [comment]
> @iPraveenParihar is this ready for review?

tests were failing before, now that PR https://github.com/rook/rook/pull/14517 is merged tests should pass.
Am waiting for CI tests to pass. Will mark it for review once CI looks 🟢

**Comment by iPraveenParihar** [comment]
seeing below logs in cephfs external-provisioner
```
W0802 14:01:24.983317       1 reflector.go:547] k8s.io/client-go/informers/factory.go:160: failed to list *v1.CSINode: csinodes.storage.k8s.io is forbidden: User "system:serviceaccount:smoke-ns-system:rook-csi-cephfs-provisioner-sa" cannot list resource "csinodes" in API group "storage.k8s.io" at the cluster scope
E0802 14:01:24.983373       1 reflector.go:150] k8s.io/client-go/informers/factory.go:160: Failed to watch *v1.CSINode: failed to list *v1.CSINode: csinodes.storage.k8s.io is forbidden: User "system:serviceaccount:smoke-ns-system:rook-csi-cephfs-provisioner-sa" cannot list resource "csinodes" in API group "storage.k8s.io" at the cluster scope
```

**Comment by iPraveenParihar** [comment]
logs from cephfs-provisioner - 

```
I0805 07:27:08.332240       1 event.go:389] "Event occurred" object="smoke-ns/file-test" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Normal" reason="Provisioning" message="External provisioner is provisioning volume for claim \"smoke-ns/file-test\""
I0805 07:27:08.332256       1 event.go:389] "Event occurred" object="smoke-ns/file-test" fieldPath="" kind="PersistentVolumeClaim" apiVersion="v1" type="Warning" reason="ProvisioningFailed" message="failed to provision volume with StorageClass \"cephfs-storageclass\": error generating accessibility requirements: no available topology found"
```

CephFS doesn't support topology based provisioning yet, but it was reporting the capability before (<v.3.11.0). I had recently removed it in https://github.com/ceph/ceph-csi/commit/8a392e5652a89e76ad7f1ed8761d669d64f35f09 and change will be in v3.12.0. Since Rook uses v3.11.0 image for CI tests hence the error.

_(fyi: earlier it was not failing because topology feature was disabled by default in external-provisioner. And now it will support topology based provisioning if CSI reports the topology capability.)_

@Madhu-1, will have to wait for v3.12.0

**Review by travisn — APPROVED** [review]

**Pull request merged**